Source/Details/ASCollectionLayoutDelegate.h

```diff
@@ -49,6 +41,19 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (ASCollectionLayoutState *)calculateLayoutWithContext:(ASCollectionLayoutContext *)context;
 
+@optional
+
+/**
+ * @abstract Returns any additional information needed for a coming layout pass with the given elements.
+ *
+ * @param elements The elements to be laid out later on.
+ *
+ * @discussion The returned object must support equality and hashing (i.e `-isEqual:` and `-hash` must be properly implemented).
+ *
+ * @discussion This method will be called on main thread.
+ */
+- (nullable id)additionalInfoForLayoutWithElements:(ASElementMap *)elements;
```

Comment:
Let's make this required – they can always return `nil` if they want, and that way they are sure to know that this method exists (and we don't have to ask the runtime whether they implement.)